src/libraries/Microsoft.PowerFx.Interpreter/Functions/CollectFunction.cs

```diff
@@ -0,0 +1,232 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Functions.FunctionArgValidators;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Types;
+using static Microsoft.PowerFx.Core.Localization.TexlStrings;
+
+namespace Microsoft.PowerFx.Interpreter
+{
+    // The CollectFunction class was copied from PowerApss.
+    // Implementation of a Set function which just chains to 
+    // RecalcEngine.UpdateVariable().
+    // Set has no return value. 
+    // Whereas PowerApps' Set() will implicitly define arg0,
+    //  this Set() requires arg0 was already defined and has a type.
+    //
+    // Called as:
+    //   Set(var,newValue)
+    internal class CollectFunction : BuiltinFunction, IAsyncTexlFunction
+    {
+        public override bool ManipulatesCollections => true;
+
+        public override bool ModifiesValues => true;
+
+        public override bool IsSelfContained => false;
```

Comment:
>public override bool IsSelfContained => false; [](http://example.com/codeflow?start=8&length=46)

why did this flag from first iteration?

Reply:
From what I can tell, this is the flag thats controls if the function can run only in a side effect context. It is actually a different flag that validates this, but it depends directly on the IsSelfContained flag.

> public virtual bool IsBehaviorOnly => !IsSelfContained;